### PR TITLE
Bug#139639 Fix errors and performance issues in Hide Children mashup

### PR DESCRIPTION
**Pull request by victor-homyakov**
- fixes JS error `TypeError: c is undefined`
- fixes backend error `Property 'NaN' does not exist in 'UserStory'`
- removes useless AJAX requests
- removes POST to client settings right after GET with the same data
- fixes JS error with paginated response
- speeds up DOM search operations

### CODE REVIEW

File: Hide Children/ChildHider.js [hunk 1]
```diff
@@ -161,54 +165,91 @@ tau.mashups
...
+            },
 
-                refreshBoardSize();
+            updateCardsVisibility: function() {
+                var $cards = $('.i-role-card');
+                var $cardsToHide = $();
+                var cardsCount = 0;
 
+                _.each(this.childrenIds, function(id) {
+                    var $cardToHide = $cards.filter('[data-entity-id=' + id + ']');
```
Comment (debrin182) [inline Hide Children/ChildHider.js]: Filter with access to data attribute is applied to elements collection. Filter by integer Id's array seem to be faster

Try someting like:
```js
var $cardToHide = _.chain($('.i-role-card')).reduce(function(aggregate, $card) {
var entityId = $card.data('entity-id');
if(this.childrenIds.indexOf(entityId) >=0)
{
     aggregate[entityId] = $card;
}
return aggregate;
}, {}).mapObject(function(card, id) {
     return card;
}).value();
```

Comment (victor-homyakov) [inline Hide Children/ChildHider.js]: Nope.

1. You'll need to wrap each collection's item in `$()` while iterating over `$cards`
2. At the end you'll need to wrap resulting array in `$()` again to convert array of DOM elements to jQuery collection

My test results show execution times: ~500ms for original code, ~700ms for proposed "faster" variant.

The last but not least: size and readability of code. Proposed fix is much harder to read and understand.

File: Hide Children/ChildHider.js [hunk 2]
```diff
@@ -36,54 +32,34 @@ tau.mashups
...
-
             init: function() {
                 var self = this;
 
-                this.cards = [];
-                this.represented = [];
-                this.state = false;
                 this.$btn = null;
                 this.boardId = 0;
+                this.acid = null;
```
Comment (victor-homyakov) [inline Hide Children/ChildHider.js]: Done

File: Hide Children/ChildHider.js [hunk 3]
```diff
@@ -36,54 +32,34 @@ tau.mashups
...
-
-        };
-
         var ChildHider = Class.extend({
-
             parentMap: {
-                'Feature': 'Epic.Id',
-                'UserStory': 'Feature.Id',
-                'Task': 'UserStory.Id'
+                Feature: {entityType: 'Epic', selector: 'Epic.Id'},
```
Comment (victor-homyakov) [inline Hide Children/ChildHider.js]: Good point

File: Hide Children/ChildHider.js (hunk 1 above)
Comment (debrin182) [inline Hide Children/ChildHider.js]: Ok, sounds reasonable. I was a bit overwhelmed during the night.  Proceed with existing code.

**Pull request merged**